src/helpers/PdfDownload/index.js

```diff
@@ -0,0 +1,38 @@
+export class PdfDownload {
+  constructor(event, form) {
+    this.submitEvt = event;
+    this.formInstance = form;
+  }
+
+  /**
+   * isPdfDownloadEnabled function check needPdf field is marked as 'yes' in form.io
+   * and then set PDF url as a session storage
+   * @return {undefined}
+   * */
+  isPdfDownloadEnabled() {
+    if (this.submitEvt.data && this.submitEvt.data.needPdf === "yes") {
+      const pdfUrl = this.submitEvt.metadata.pdfUrl.DownloadUrl;
+      window.sessionStorage.setItem("pdfUrl", pdfUrl);
+      this.formInstance.root.nextPage().then(() => {
+        document.getElementsByClassName(
+          "formio-wizard-nav-container"
+        )[0].style.visibility = "hidden";
+      });
+      this.onDownloadbtnClick();
```

Comment:
i don't think you'll need a global click event if you stick to the implementation of the example in https://api.forms.platforms.qld.gov.au/#/project/62737be5449895454f54cd93/form/62737be6449895454f54cdd7/edit

where you can set `data.pdfUrl = s.metadata.pdfUrl.DownloadUrl;` here
and the button in form.io to be `<a href="{{data.pdfUrl}}" class="btn btn-primary">Download</a>`

also you don't need to check `this.submitEvt.data.needPdf` as you can check `data.pdfUrl` instead

Reply:
this is not working well from an external file because of loading time etc. I tried, it works from inside forms.io but not from repository files

With this one, I need to add data back in the forms.io data object. I tried from the repository but race conditions and sometimes I get that appended sometimes it takes more time